### ISSUE TITLE
SDSS Query returns OverflowError: Python int too large to convert to C long

### ISSUE THREAD

**Issue by astro-fermi** [astropy/astroquery#2839]
I am trying to run an SDSS query using a list of coordinates. Sometimes, it works but sometimes it fails. Below I show an example:

from astroquery.sdss import SDSS
from astropy import coordinates
from astropy import units as u
pos = coordinates.SkyCoord(ra=2.023458 *u.deg, dec=14.8398 *u.deg, frame='icrs')
xid = SDSS.query_region(pos, radius='5 arcsec', spectro=True)
print(xid)
       ra              dec               objid        run  ... fiberID     specobjid      run2d
---------------- ---------------- ------------------- ---- ... ------- ------------------ -----
2.02344596573482 14.8398237551311 1237652943176138868 1739 ...     160 845594848269461504    26

pos = coordinates.SkyCoord(ra=216.231298 *u.deg, dec=36.260011 *u.deg, frame='icrs')
xid = SDSS.query_region(pos, radius='5 arcsec', spectro=True)
WARNING: OverflowError converting to IntType in column specobjid, reverting to String. [astropy.io.ascii.fastbasic]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/miniconda3/lib/python3.8/site-packages/astroquery/utils/class_or_instance.py", line 25, in f
    return self.fn(obj, *args, **kwds)
  File "/opt/miniconda3/lib/python3.8/site-packages/astroquery/utils/process_asyncs.py", line 29, in newmethod
    result = self._parse_result(response, verbose=verbose)
  File "/opt/miniconda3/lib/python3.8/site-packages/astroquery/sdss/core.py", line 948, in _parse_result
    arr[id_column] = arr[id_column].astype(np.int64)
OverflowError: Python int too large to convert to C long

I don't know if others have faced the same issue. Please help.

**Comment by bsipocz** [comment]
cc @hamogu as the OverFlowError seems to be coming from astropy's ascii reader.

Would you recommend using a pure python reader instead of using a fast reader?

**Comment by hamogu** [comment]
First, let me say that your  example works for me without warning or error. Without being able to reproduce the error it's very hard to debug for certain, so everything I say below is somewhat of a speculation:
```Python
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:41:52) [Clang 15.0.7 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.14.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from astroquery.sdss import SDSS
   ...: from astropy import coordinates
   ...: from astropy import units as u
   ...: pos = coordinates.SkyCoord(ra=2.023458 *u.deg, dec=14.8398 *u.deg, frame='icrs')
   ...: xid = SDSS.query_region(pos, radius='5 arcsec', spectro=True)
   ...: 

In [2]: print(xid)
       ra              dec               objid        run  rerun camcol field     z      plate  mjd  fiberID     specobjid      run2d
---------------- ---------------- ------------------- ---- ----- ------ ----- ---------- ----- ----- ------- ------------------ -----
2.02344596573482 14.8398237551311 1237652943176138868 1739   301      3   315 0.04559058   751 52251     160 845594848269461504    26

In [3]: pos = coordinates.SkyCoord(ra=216.231298 *u.deg, dec=36.260011 *u.deg, frame='icrs')
   ...: xid = SDSS.query_region(pos, radius='5 arcsec', spectro=True)
   ...: 

In [4]: print(xid)
       ra             dec               objid        run  rerun camcol field    z     plate  mjd  fiberID      specobjid      run2d
--------------- ---------------- ------------------- ---- ----- ------ ----- -------- ----- ----- ------- ------------------- -----
216.23128680229 36.2600114173511 1237662225688952867 3900   301      5   540 6.041818  1644 53144     452 1851103744410806272    26
```
In general, @bsipocz is correct: If the io.ascii C implementation has a problem, then it's always a good idea to try the Python reader. However, I don't think that it's the `astropy.io.ascii` reader that is the problem. That reader encounters a problem, issues a warning, and describes how it is working around that problem ("WARNING: OverflowError converting to IntType in column specobjid, reverting to String.") Now, as you see in my output, I don't get that warning either, so I can't even say for sure why you get that.

@astro-fermi : What type of system are you running on? Is that a 32-bit system?

The traceback in fact points to a line in astroquery:
```Python
File "/opt/miniconda3/lib/python3.8/site-packages/astroquery/sdss/core.py", line 948, in _parse_result
arr[id_column] = arr[id_column].astype(np.int64)
OverflowError: Python int too large to convert to C long
```
where we see the same problem that the io.ascii reader encountered ("int too large to convert to C long"), but this time it's fatal and throws an error and not a warning.

So, why does @astro-fermi see that and I don't? I see two possibilities: Are you on a 32 bit system or some unusual architecture (powerpc, linux ARM, etc.) where a C long is different from what we see?

Or maybe you are running into some bug that's been fixed a while ago. I see you are on Python 3.8, while I'm running 3.10. Similarly, that means that your numpy and your astroquery are probably older than my versions. Can you tell us which versions you have?
Just run the code below and paste the output here:
```Python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import astropy; print("astropy", astropy.__version__)
import numpy; print("Numpy", numpy.__version__)
import astroquery; print("astroquery", astroquery.__version__)
```

For example, the environment where I tested gives:
```
macOS-13.2.1-arm64-arm-64bit
Python 3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:41:52) [Clang 15.0.7 ]
astropy 5.1
Numpy 1.25.2
astroquery 0.4.5
```

One of the easiest things to try is for you to make a new environment using current versions and see if the problem still occurs.

**Comment by astro-fermi** [comment]
Hi All,

Thanks for the quick suggestions. I use MAC OS Ventura. I guess it is a 64-bit system. Following your suggestion, I created a new conda environment with Python 3.11 and installed astropy, numpy, astroquery, etc. The same code started running without any problem. For curiosity, I also tried running the same code once again using the previous Python 3.8 and now there is no error at all. Regarding the architecture, the following is the output:

macOS-10.16-x86_64-i386-64bit
Python 3.8.6 | packaged by conda-forge | (default, Dec 26 2020, 04:50:20) 
[Clang 11.0.0 ]
astropy 5.2.2
Numpy 1.24.4
astroquery 0.4.7.dev8479

Since the problem is solved, the thread can be closed.
Thank you once again for all the help.
